dev-lang/lua/lua-5.4.2-r2.ebuild

```
@@ -0,0 +1,199 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit autotools multilib multilib-minimal portability toolchain-funcs
+
+DESCRIPTION="A powerful light-weight programming language designed for extending applications"
+HOMEPAGE="https://www.lua.org/"
+TEST_PV="5.4.2"
+TEST_P="${PN}-${TEST_PV}-tests"
+SRC_URI="
+	https://www.lua.org/ftp/${P}.tar.gz
+	test? ( https://www.lua.org/tests/${TEST_P}.tar.gz )"
+
+LICENSE="MIT"
+SLOT="5.4"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+IUSE="+deprecated readline test test-complete"
+
+COMMON_DEPEND="
+	>=app-eselect/eselect-lua-3
+	readline? ( sys-libs/readline:0= )
+	!dev-lang/lua:0"
+DEPEND="
+	${COMMON_DEPEND}
+	sys-devel/libtool"
+RDEPEND="${COMMON_DEPEND}"
+
+RESTRICT="!test? ( test )"
+
+MULTILIB_WRAPPED_HEADERS=(
+	/usr/include/lua${SLOT}/luaconf.h
+)
+
+PATCHES=(
+	"${FILESDIR}"/lua-5.4.2-make.patch
+)
+
+src_prepare() {
+	default
+	# use glibtool on Darwin (versus Apple libtool)
+	if [[ ${CHOST} == *-darwin* ]] ; then
+		sed -i -e '/LIBTOOL = /s:/libtool:/glibtool:' \
+			Makefile src/Makefile || die
+	fi
+
+	# correct lua versioning
+	sed -i -e 's/\(LIB_VERSION = \)6:1:1/\10:0:0/' src/Makefile || die
+
+	sed -i -e 's:\(/README\)\("\):\1.gz\2:g' doc/readme.html || die
+
+	# Using dynamic linked lua is not recommended for performance
+	# reasons. http://article.gmane.org/gmane.comp.lang.lua.general/18519
+	# Mainly, this is of concern if your arch is poor with GPRs, like x86
+	# Note that this only affects the interpreter binary (named lua), not the lua
+	# compiler (built statically) nor the lua libraries.
+
+	# upstream does not use libtool, but we do (see bug #336167)
+	cp "${FILESDIR}/configure.in" "${S}/configure.ac" || die
+	eautoreconf
+
+	# custom Makefiles
+	multilib_copy_sources
+}
+
+multilib_src_configure() {
+	sed -i \
+		-e 's:\(define LUA_ROOT\s*\).*:\1"'${EPREFIX}'/usr/":' \
+		-e "s:\(define LUA_CDIR\s*LUA_ROOT \"\)lib:\1$(get_libdir):" \
+		src/luaconf.h \
+	|| die "failed patching luaconf.h"
+
+	econf
+}
+
+multilib_src_compile() {
+	tc-export CC
+
+	# what to link to liblua
+	liblibs="-lm"
+	liblibs="${liblibs} $(dlopen_lib)"
+
+	# what to link to the executables
+	mylibs=
+	use readline && mylibs="-lreadline"
+
+	cd src
+
+	local myCFLAGS=""
+	use deprecated && myCFLAGS+="-DLUA_COMPAT_5_3 "
+	use readline && myCFLAGS+="-DLUA_USE_READLINE "
+
+	case "${CHOST}" in
+		*-mingw*) : ;;
+		*) myCFLAGS+="-DLUA_USE_LINUX " ;;
+	esac
+
+	emake CC="${CC}" CFLAGS="${myCFLAGS} ${CFLAGS}" \
+			SYSLDFLAGS="${LDFLAGS}" \
+			RPATH="${EPREFIX}/usr/$(get_libdir)/" \
+			LUA_LIBS="${mylibs}" \
+			LIB_LIBS="${liblibs}" \
+			V=$(ver_cut 1-2) \
+			LIBTOOL="${EROOT}/usr/bin/libtool" \
```

```suggestion
			LIBTOOL="${ESYSROOT}/usr/bin/libtool" \
```
`EROOT` is not what you think it is (it's only valid in `pkg_*` phases)